service/src/test/java/bio/terra/catalog/service/dataset/DatasetDaoTest.java

```diff
@@ -0,0 +1,72 @@
+package bio.terra.catalog.service.dataset;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.catalog.app.App;
+import bio.terra.catalog.common.StorageSystem;
+import bio.terra.catalog.service.dataset.exception.InvalidDatasetException;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ContextConfiguration(classes = App.class)
+class DatasetDaoTest {
+
+  @Autowired private DatasetDao datasetDao;
+  @Autowired private NamedParameterJdbcTemplate jdbcTemplate;
+
+  private Dataset createDataset(String datasetId, StorageSystem storageSystem, String metadata)
+      throws InvalidDatasetException {
+    Dataset dataset = new Dataset(null, datasetId, storageSystem, metadata, null);
+    return datasetDao.create(dataset);
+  }
+
+  @Test
+  void testDatasetCrudOperations() {
+    String datasetId = UUID.randomUUID().toString();
+    String metadata = "{\"sampleId\": \"12345\", \"species\": [\"mouse\", \"human\"]}";
```

Comment:
It looks like this is the same for all tests; if so, I would make it a static final field in the test class. Also JDK 17 includes string blocks, so you could write this as:
```suggestion
    String metadata = """
                                  {"sampleId": "12345", "species": ["mouse", "human"]}""";
```